In systemd the samba service refers to samba4 Active Directory services
For this project we need smbd instead

Patch by Gordon Bos

--- a/web-admin/bin/adminfunctions.php
+++ b/web-admin/bin/adminfunctions.php
@@ -416,7 +416,7 @@
 		if(query_service("avahi-daemon")){
 			restart_service("avahi-daemon");
 		}
-		if(query_service("samba")){
+		if(query_service("smbd")){
 			restart_samba();
 		}
 		if(query_service("forked-daapd")){
--- a/web-admin/lib/Bubba.pm
+++ b/web-admin/lib/Bubba.pm
@@ -498,7 +498,7 @@
 # Return : Status of operation.
 
 sub restart_samba {
-   system("/usr/bin/systemctl", "restart", "samba");
+   system("/usr/bin/systemctl", "restart", "smbd");
 
    return $?;
 }
@@ -512,7 +512,7 @@
 # Return : Status of operation.
 
 sub reload_samba {
-   system("/usr/bin/systemctl", "reload", "samba");
+   system("/usr/bin/systemctl", "reload", "smbd");
 
    return $?;
 }
@@ -1422,7 +1422,7 @@
 	# services, boolean such if service enabled or not
 	my %services = map {
 		$_ => (system("systemctl is-enabled $_")==0);
-	} qw(proftpd forked-daapd ntpd filetransferdaemon cupsd postfix dovecot fetchmail minidlna dnsmasq logitechmediaserver hostapd netatalk net.br0 samba);
+	} qw(proftpd forked-daapd ntpd filetransferdaemon cupsd postfix dovecot fetchmail minidlna dnsmasq logitechmediaserver hostapd netatalk smbd);
 
 	my $meta = {
 		version => $revision,
@@ -1658,7 +1658,7 @@
 		}
 		$lan = _get_lanif;
 		restart_network($lan);
-		reload_service("samba");
+		reload_service("smbd");
 		# Dont reload apache for now, it breaks connection with fcgi
 		# Resulting in an internal server error.
 		#reload_service("apache2");
@@ -1840,7 +1840,7 @@
 				remove_service("logitechmediaserver");
 			}
 
-			reload_service("samba");
+			reload_service("smbd");
 
 			# Dont reload apache for now, it breaks connection with fcgi
 			# Resulting in an internal server error.
Only in b/web-admin/lib: Bubba.pm.orig
